StreamReader.h

```diff
@@ -32,12 +39,14 @@ class FileStreamReader : public SeekableStreamReader {
 
 class MemoryStreamReader : public SeekableStreamReader {
 public:
-	MemoryStreamReader(char* buffer, size_t size);
-	void Read(char* buffer, size_t size);
-	void SeekRelative(int offset);
+	MemoryStreamReader(char* buffer, uint64_t size);
```

Comment:
I'm a bit on the fence here, but size could also be of type `size_t` here too, at least for the constructor parameter. The implementation of this class is necessarily restricted to memory size, and the constructor will always be implementation specific. (As in, it can never be virtual).

Virtual method signatures will of course have to match the parent class, so they will need to take `uint64_t`. The return value of virtual methods must be covariant (roughly meaning the same, or a subclass of), which might allow `size_t` as the return type, though I'm not sure that's guaranteed to be covariant with `uint64_t` on all platforms.

Internal field sizes could potentially be either `size_t` or `uint64_t`. I suppose it depends on casting requirements and convenience.

We could revisit this later as a low priority item.